src/utils/cycles.ts

```diff
@@ -0,0 +1,62 @@
+export type DependenciesFunction<T> = (t: T) => T[]
+
+
+/**
+ * Compute the dependency order of the given `things`, computing dependencies with the given `dependenciesFunction`.
```

Comment:
Could you provide an example of where this could be used? I think that would help clarify the goal of this as it is a bit abstract